### PR TITLE
v11(docs): spacing page updates

### PR DESCRIPTION
**Pull request by aagonzales**
Closes https://github.com/carbon-design-system/carbon/issues/10975

- cleaned-up structure of spacing page
- added section about the "stack" component

### CODE REVIEW

**Comment by aagonzales** [comment]
@tay1orjones Is this enough for `stack` on the overview page?

File: src/pages/guidelines/spacing/overview.mdx [hunk 1]
```diff
@@ -93,6 +110,32 @@ Each has its own unique use case.
...
+property which will allow a user to provide a custom value for the gap of the
+layout.
+
+For more information, see the `Stack` component in
+[React storybook](https://carbon-react-next.netlify.app/?path=/story/layout-stack--default).
+
+<Row>
+<Column colLg={8}>
+
+![Example of spacing tokens applied with the stack component](images/spacing-stack.png.png)
```
Comment (tay1orjones) [inline src/pages/guidelines/spacing/overview.mdx:132]:
```suggestion
![Example of spacing tokens applied with the stack component](images/spacing-stack.png)
```

**Review by tay1orjones — CHANGES_REQUESTED** [review]
@aagonzales Yes, fantastic! One small change needed to the image path to make sure it'll render though.

**Review by tay1orjones — APPROVED** [review]

**Review by joshblack — APPROVED** [review]

**Review by laurenmrice — APPROVED** [review]
Looks great! 🚀

**Pull request merged**